src/library_pthread.js

```diff
@@ -416,6 +422,8 @@ var LibraryPThread = {
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 20) >> 2, threadParams.schedPolicy);
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 24) >> 2, threadParams.schedPrio);
 
+    Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.locale }}}) >> 2, __global_locale_ptr);
```

Comment:
This already lives on the shared heap by virtue of originally coming from `libc.global_locale`. We still need an Emscripten-specific setup function, but we can localize it to `library_pthread.c` and call that from `library_pthread.js`, which seems reasonable to me?

PTAL: d5fe3471cb49403047896db40e790e4c1ccabf5a